src/ngAnimate/animateQueue.js

```diff
@@ -0,0 +1,558 @@
+          }
+        });
+      }
+
+      if (isGeneratedID) {
+        element.removeAttr('id');
+      }
+
+      return matches;
+    }
+
+    // $animate.on('enter', element, function() {
+    //
+    // });
+    // $animate.on('enter', '[ng-view]', function() {
+    //
+    // });
+    // element.on('$animate:enter', j
+    function triggerCallback(event, element, phase, data) {
+      $$rAF(function() {
+        forEach(findCallbacks(element, event), function(callback) {
+          callback(element, phase, data);
+        });
+      });
+    }
+
+    return {
+      on: function(event, container, callback) {
+        var node = extractElementNode(container);
+        callbackRegistry[event] = callbackRegistry[event] || [];
+        callbackRegistry[event].push({
+          node: node,
+          callback: callback
+        });
+      },
+
+      off: function(event, container, callback) {
+        var entries = callbackRegistry[event];
+        if (!entries) return;
+
+        switch (arguments.length) {
+          case 1:
+            callbackRegistry[event] = null;
+            break;
+
+          case 2:
+            callbackRegistry[event] = filterFromRegistry(entries, container);
+            break;
+
+          case 3:
+            callbackRegistry[event] = filterFromRegistry(entries, container, callback);
+            break;
+        }
+
+        function filterFromRegistry(list, matchContainer, matchCallback) {
+          var containerNode = matchContainer[0];
+          return list.filter(function(entry) {
+            var isMatch = entry.node === containerNode &&
+                            (!matchCallback || entry.callback === matchCallback);
+            return !isMatch;
+          });
+        }
+      },
+
+      push: function(element, event, options, domOperation) {
+        element = stripCommentsFromElement(element);
+        options = options || {};
+        options.domOperation = domOperation;
+        return queueAnimation(element, event, options);
+      },
+
+      enabled: function(bool) {
+        var argCount = arguments.length;
+        if (isElement(bool)) {
+          var element = bool;
+          var node = element.length ? element[0] : element;
+          var recordExists = disabledElementsLookup.get(node);
+
+          // if nothing is set then the animation is enabled
+          bool = !recordExists;
+
+          if (argCount > 1) {
+            bool = !!arguments[1];
+            if (!bool) {
+              disabledElementsLookup.put(node, true);
+            } else if (recordExists) {
+              disabledElementsLookup.remove(node);
+            }
+          }
+        } else if (argCount === 1) {
+          bool = animationsEnabled = !!bool;
+        } else {
+          bool = animationsEnabled;
+        }
+
+        return bool === true;
+      }
+    };
+
+    function queueAnimation(element, event, options) {
+      options = $$animateOptions(element, options);
+      var node = element[0];
```

Comment:
Basically, I see `element[0]` is used most of the time. I wonder if it may introduce subtle bugs.
Would it make sense to replace all occurences of `element[0]` with `extractElementNode(element)` or is it too much overhead ?